### PR TITLE
Port trajectory_rviz_plugin

### PR DESCRIPTION
**Pull request by RoboticsYY**
### Description

This PR intends to port the `moveit_ros_visualization/trajectory_rviz_plugin` to ROS2. The dependencies were ported previously in PR #160.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)

### CODE REVIEW

**Comment by RoboticsYY** [comment]
> I opened a PR to enable the trajectory visualization in the demo [RoboticsYY#1](https://github.com/RoboticsYY/moveit2/pull/1)

@JafarAbdi Thanks for helping change the MoveItCpp demo!

File: moveit_ros/visualization/trajectory_rviz_plugin/include/moveit/trajectory_rviz_plugin/trajectory_display.h [hunk 1]
```diff
@@ -72,7 +74,7 @@ class TrajectoryDisplay : public rviz::Display
   void onInitialize() override;
   void onEnable() override;
   void onDisable() override;
-  void setName(const QString& name) override;
+  // void setName(const QString& name) override;
```
[the remark this answers is not in the document]
Comment (RoboticsYY) [reply moveit_ros/visualization/trajectory_rviz_plugin/include/moveit/trajectory_rviz_plugin/trajectory_display.h]: I didn't figure out why Rviz2 made this method `final`. But I think removing this method will not affect the functionality of this plugin. Done remove.

File: moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp [hunk 2]
```diff
@@ -145,6 +147,9 @@ void TrajectoryVisualization::onInitialize(const rclcpp::Node::SharedPtr& node,
   context_ = context;
   node_ = node;
 
+  auto ros_node_abstraction = context_->getRosNodeAbstraction().lock();
```
Comment (henningkayser) [inline moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp:150]: Do we need to check if the weak_ptr lock was successful?

File: moveit_ros/visualization/trajectory_rviz_plugin/src/trajectory_display.cpp [hunk 3]
```diff
@@ -58,21 +58,23 @@ void TrajectoryDisplay::onInitialize()
 {
   Display::onInitialize();
 
-  trajectory_visual_->onInitialize(scene_node_, context_, update_nh_);
+  node_ = context_->getRosNodeAbstraction().lock()->get_raw_node();
```
Comment (henningkayser) [inline moveit_ros/visualization/trajectory_rviz_plugin/src/trajectory_display.cpp:61]: same

File: moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp (hunk 2 above)
Comment (RoboticsYY) [reply moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp:150]: Yes, it's better to make sure the lock() return non-empty pointer. A check has been added.

File: moveit_ros/visualization/trajectory_rviz_plugin/src/trajectory_display.cpp (hunk 3 above)
Comment (RoboticsYY) [reply moveit_ros/visualization/trajectory_rviz_plugin/src/trajectory_display.cpp]: Added check here as well.

**Pull request merged**